### ISSUE TITLE
[no-useless-constructor] ignores parameter properties

### ISSUE THREAD

**Issue by texastoland** [typescript-eslint/typescript-eslint#334]
**Repro**

```JSON
{
  "rules": {
    "@typescript/no-useless-constructor": "error"
  }
}
```

```TS
class C {
  constructor(readonly x) {}
}
```

**Expected Result**

No error.

**Actual Result**

`Useless constructor.eslint(no-useless-constructor)`

**Additional Info**

#167 @armano2

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.2.0` |
| `@typescript-eslint/parser`        | `1.4.2` |
| `TypeScript`                       | `3.3.3333` |
| `ESLint`                           | `5.15.1` |
| `node`                             | `6.7.0` |
| `npm`                              | `11.1.0` |

**Comment by bradzacher** [comment]
not going to lie, I hate that that is valid syntax.

**Comment by texastoland** [comment]
FP-ers don't but there's a separate rule for that: `no-parameter-properties`.

**Comment by bradzacher** [comment]
I never liked parameter properties, but what I mean is that I hate that you don't need to specify a modifier for a parameter property.

I just tried testing against master, could you please try updating to 1.4.2 and report if that's working?

**Comment by texastoland** [comment]
> Could you please try updating to 1.4.2 and report if that's working?

That what was in my `yarn.lock`.

> I just tried testing against master and it's working fine.

I see what you mean 🤔
https://github.com/armano2/typescript-eslint/blob/76600a53e0695b3d1051cb6e6a2b8802631438c2/packages/eslint-plugin/lib/rules/no-useless-constructor.js#L45

I made a new test repo and realized it was the JS rule thanks for checking!
